File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeDomainEventPublisher.java
package com.study.platform.support.fake;

import com.study.platform.global.event.DomainEventPublisher;

import java.util.ArrayList;
import java.util.List;

public class FakeDomainEventPublisher implements DomainEventPublisher {

    private final List<Object> events = new ArrayList<>();

    @Override
    public void publish(Object event) {
        events.add(event);
    }

    public List<Object> getEvents() {
        return events;
    }

    // 특정 타입 이벤트만 꺼낼 때
    public <T> List<T> getEventsOf(Class<T> type) {
        return events.stream()
                .filter(type::isInstance)
                .map(type::cast)
                .toList();
    }

    public boolean hasEventOf(Class<?> type) {
        return events.stream().anyMatch(type::isInstance);
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeDomainEventPublisher.kt
package com.study.platform.support.fake

import com.study.platform.global.event.DomainEventPublisher

class FakeDomainEventPublisher : DomainEventPublisher {

    private val events: MutableList<Any> = ArrayList()

    override fun publish(event: Any) {
        events.add(event)
    }

    fun getEvents(): List<Any> = events

    fun <T> getEventsOf(type: Class<T>): List<T> =
        events.filterIsInstance(type)

    fun hasEventOf(type: Class<*>): Boolean =
        events.any { type.isInstance(it) }
}
